pkg/reconciler/v1alpha1/serverlessservice/serverlessservice.go

```diff
@@ -245,13 +246,28 @@ func (r *reconciler) reconcilePublicService(ctx context.Context, sks *netv1alpha
 func (r *reconciler) reconcilePublicEndpoints(ctx context.Context, sks *netv1alpha1.ServerlessService) error {
 	logger := logging.FromContext(ctx)
 
-	// Service and Endpoints have the same name.
-	// Get private endpoints first, since if they are not available there's nothing we can do.
-	psn := names.PrivateService(sks.Name)
-	srcEps, err := r.endpointsLister.Endpoints(sks.Namespace).Get(psn)
-	if err != nil {
-		logger.Errorw(fmt.Sprint("Error obtaining private service endpoints:", psn), zap.Error(err))
-		return err
+	var (
+		srcEps *corev1.Endpoints
+		err    error
+	)
+	switch sks.Spec.Mode {
+	case netv1alpha1.SKSOperationModeServe:
+		// Service and Endpoints have the same name.
+		// Get private endpoints first, since if they are not available there's nothing we can do.
+		psn := names.PrivateService(sks.Name)
+		srcEps, err = r.endpointsLister.Endpoints(sks.Namespace).Get(psn)
+		if err != nil {
+			logger.Errorw(fmt.Sprintf("Error obtaining private service endpoints: %s", psn), zap.Error(err))
+			return err
+		}
+		logger.Debugf("Private endpoints: %s", spew.Sprint(srcEps))
+	case netv1alpha1.SKSOperationModeProxy:
+		srcEps, err = r.endpointsLister.Endpoints(system.Namespace()).Get(activatorService)
+		if err != nil {
+			logger.Errorw("Error obtaining activator service endpoints", zap.Error(err))
+			return err
+		}
+		logger.Debugf("Activator endpoints: %s", spew.Sprint(srcEps))
 	}
```

Comment:
`default:`?

Reply:
this is a enum and we validate the value in WH, so no other option for now.